numpy/core/src/private/ufunc_override.c

```diff
@@ -0,0 +1,532 @@
+         */
+        if (has_non_default_array_ufunc(obj)) {
+            if (with_override != NULL) {
+                with_override[noa] = obj;
+            }
+	    ++noa;
+        }
+    }
+    return noa;
+
+fail:
+    return -1;
+}
+/*
+ * Check a set of args for the `__array_ufunc__` method.  If more than one of
+ * the input arguments implements `__array_ufunc__`, they are tried in the
+ * order: subclasses before superclasses, otherwise left to right. The first
+ * (non-None) routine returning something other than `NotImplemented`
+ * determines the result. If all of the `__array_ufunc__` operations return
+ * `NotImplemented` (or are None), a `TypeError` is raised.
+ *
+ * Returns 0 on success and 1 on exception. On success, *result contains the
+ * result of the operation, if any. If *result is NULL, there is no override.
+ */
+NPY_NO_EXPORT int
+PyUFunc_CheckOverride(PyUFuncObject *ufunc, char *method,
+                      PyObject *args, PyObject *kwds,
+                      PyObject **result,
+                      int nin)
+{
+    int i;
+    int j;
+
+    int noa;
+    PyObject *with_override[NPY_MAXARGS];
+
+    PyObject *obj;
+    PyObject *other_obj;
+
+    PyObject *method_name = NULL;
+    PyObject *normal_args = NULL; /* normal_* holds normalized arguments. */
+    PyObject *normal_kwds = NULL;
+
+    PyObject *override_args = NULL;
+    Py_ssize_t len;
+
+    /*
+     * Check inputs for overrides
+     */
+    noa = PyUFunc_HasOverride(args, kwds, with_override);
+    /* No overrides, bail out.*/
+    if (noa == 0) {
+        *result = NULL;
+        return 0;
+    }
+
+    /*
+     * Normalize ufunc arguments.
+     */
+
+    /* Build new kwds */
+    if (kwds && PyDict_CheckExact(kwds)) {
+        PyObject *out;
+
+        /* ensure out is always a tuple */
+        normal_kwds = PyDict_Copy(kwds);
+        out = PyDict_GetItemString(normal_kwds, "out");
+        if (out != NULL) {
+            if (PyTuple_Check(out)) {
+                int all_none = 1;
+                int i;
+
+                for (i = 0; i < PyTuple_GET_SIZE(out); i++) {
+                    all_none = (PyTuple_GET_ITEM(out, i) == Py_None);
+                    if (!all_none) {
+                        break;
+                    }
+                }
+                if (all_none) {
+                    PyDict_DelItemString(normal_kwds, "out");
+                }
+            }
+            else if (out != Py_None) {
+                /* not already a tuple and not None */
+                PyObject *out_tuple = PyTuple_New(1);
+
+                if (out_tuple == NULL) {
+                    goto fail;
+                }
+                /* out was borrowed ref; make it permanent */
+                Py_INCREF(out);
+                /* steals reference */
+                PyTuple_SET_ITEM(out_tuple, 0, out);
+                PyDict_SetItemString(normal_kwds, "out", out_tuple);
+                Py_DECREF(out_tuple);
+            }
```

Comment:
I think that with the current normalization, this needs an `else` to `PyDict_DelItemString(normal_kwds, "out");`